### PR TITLE
HOPSWORKS-329

### PR DESCRIPTION
**Pull request by kouzant**
* Add project generic user as owner of notebook

* Allow every member to move notes to Trash but only the Data Owner can empty the Trash

* Log activity of moving Notebooks to Zeppelin Trash

* Change Zeppelin Empty trash icon :)

* Zeppelin UI handles the error message. Hopsworks should handle it somehow...

* Handle error message in Hopsworks UI

* Change log severity for exception thrown when user does not have permissions to empty the Trash bin

### CODE REVIEW

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/zeppelin/notebook/repo/HDFSNotebookRepo.java [hunk 1]
```diff
@@ -55,6 +58,9 @@
   private final Configuration hdfsConf;
   private final DistributedFsService dfsService;
   private final UserGroupInformation superuser;
+  private final Pattern psuPattern = Pattern.compile("(\\w*)__\\w*");
```
Comment (tkakantousis) [inline hopsworks-api/src/main/java/io/hops/hopsworks/api/zeppelin/notebook/repo/HDFSNotebookRepo.java]: @kouzant can we get the "__" characters from a constant, like Setting.java?

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/zeppelin/notebook/repo/HDFSNotebookRepo.java [hunk 2]
```diff
@@ -357,24 +363,42 @@ public synchronized void save(Note note, AuthenticationInfo subject) throws
...
-      if (!udfso.getFilesystem().isDirectory(noteDir)) {
+      if (!dfso.getFilesystem().isDirectory(noteDir)) {
         // it does not look like zeppelin note savings
         throw new IOException("Can not remove " + noteDir.toString());
       }
-      udfso.getFilesystem().delete(noteDir, true);
+      dfso.getFilesystem().delete(noteDir, true);
     } finally {
-      closeDfsClient(udfso);
+      if (dfso != null) {
```
Comment (tkakantousis) [inline hopsworks-api/src/main/java/io/hops/hopsworks/api/zeppelin/notebook/repo/HDFSNotebookRepo.java]: @kouzant Just to be more coherent, although for dfso it makes no difference, wouldn't it be better to still use closeDfsClient()?

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/zeppelin/socket/NotebookServer.java [hunk 3]
```diff
@@ -157,52 +157,59 @@ public void onMessage(String msg, Session conn) {
     try {
       Message messagereceived = deserializeMessage(msg);
       LOG.log(Level.FINE, "RECEIVE << {0}", messagereceived.op);
-      LOG.log(Level.FINE, "RECEIVE PRINCIPAL << {0}", messagereceived.principal);
+      LOG.log(Level.FINE, "RECEIVE PRINCIPAL << {0}",
```
Comment (tkakantousis) [inline hopsworks-api/src/main/java/io/hops/hopsworks/api/zeppelin/socket/NotebookServer.java]: @kouzant can you undo this refactoring? Set the right margin to 120 on your IDE.

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/zeppelin/notebook/repo/HDFSNotebookRepo.java (hunk 2 above)
Comment (kouzant) [reply hopsworks-api/src/main/java/io/hops/hopsworks/api/zeppelin/notebook/repo/HDFSNotebookRepo.java]: At the time there was a bug, fixed by https://github.com/hopshadoop/hopsworks/pull/762
Updated the code also here.

**Pull request merged**